client/me/account/main.jsx

```diff
@@ -555,6 +561,18 @@ const Account = createReactClass( {
 						</FormFieldset>
 					) }
 
+				<FormFieldset>
+					<FormLabel htmlFor="">{ translate( 'Usage Statistics' ) }</FormLabel>
+					<FormToggle
+						disabled={ this.getDisabledState() }
+						id={ isSendingTracksEventsKey }
+						checked={ userSettings.isSendingTracksEvents() }
+						onChange={ this.updateIsSendingTracksEvents }
```

Comment:
This appears to cause the cookie to be dropped before the settings are saved -- maybe edge-case, but should we consider this and only do the update on save?

Reply:
Yes I found that one as well ;) . I don't think it's a big issue to have the cookie (un)set before the setting being saved server-side if I balance it with the added complexity of all the solution I could find to fix it.

In an ideal world, I would directly save on toggle but it wouldn't be consistent UX-wise with the rest of the settings.

In a slightly less ideal world, I would love to set the cookie only when clicking on save. But since we rely on the `change` event being emitted from `user-settings` to set the cookie, we're left only with poor solutions to properly trigger the (un)set cookie side effect:
- we don't emit a `change` when we do an `updateSetting` for the opt-out, and we let `saveSettings` emit its `change`.
- we store the state of the checkbox in the `account/main` React component and `updateSetting` on submit.
- we add ( or update `change` ) an event in `lib/user-settings` representing whether or not the change was server-backed or if it's only a client-side change.

All other settings work fine because they don't have that pesky side-effect to deal with.